src/main/java/uk/gov/cslearning/catalogue/api/CourseController.java

```diff
@@ -144,9 +150,80 @@ public CourseController(CourseRepository courseRepository, ResourceRepository re
     public ResponseEntity<Module> getModule(@PathVariable String courseId, @PathVariable String moduleId) {
         LOGGER.debug("Getting module {} of course {}", moduleId, courseId);
 
+        if (!courseRepository.existsById(courseId)) {
+            return ResponseEntity.badRequest().build();
+        }
+
         Optional<Module> result = moduleService.find(courseId, moduleId);
 
         return result.map(module -> new ResponseEntity<>(module, OK))
                 .orElseGet(() -> new ResponseEntity<>(NOT_FOUND));
     }
+
+    @PostMapping("/{courseId}/modules/{moduleId}/events")
+    public ResponseEntity<Object> createEvent(@PathVariable String courseId, @PathVariable String moduleId, @RequestBody Event event, UriComponentsBuilder builder){
+        LOGGER.debug("Adding event to module with ID {}", moduleId);
+
+        if (!courseRepository.existsById(courseId)) {
+            return ResponseEntity.badRequest().build();
+        }
+
+        Event saved = eventService.save(courseId, moduleId, event);
+
+        LOGGER.info("Saved event {}", saved.toString());
+
+        return ResponseEntity.created(builder.path("/courses/{courseId}/modules/{moduleId}/events/{eventId}").build(courseId, moduleId, saved.getId())).build();
+    }
+
+    @GetMapping("/{courseId}/modules/{moduleId}/events/{eventId}")
+    public ResponseEntity<Event> getEvent(@PathVariable String courseId, @PathVariable String moduleId, @PathVariable String eventId) {
+        LOGGER.debug("Getting event {} of module{} of course {}", eventId, moduleId, courseId);
+
+        Optional<Event> result = eventService.find(courseId, moduleId, eventId);
+
+        return result.map(event -> new ResponseEntity<>(event, OK)).orElseGet(() -> new ResponseEntity<>(NOT_FOUND));
+    }
+
+    @PutMapping("/{courseId}/modules/{moduleId}/events/{eventId}")
+    public ResponseEntity<Event> updateEvent(@PathVariable String courseId, @PathVariable String moduleId, @PathVariable String eventId, @RequestBody Event newEvent){
+        LOGGER.debug("Updating event with id {}", eventId);
+
+        if(!courseRepository.existsById(courseId)){
+            return ResponseEntity.badRequest().build();
+        }
+
+        Optional<Course> result = courseRepository.findById(courseId);
+
+        return result.map(course -> {
+            FaceToFaceModule module = (FaceToFaceModule) course.getModuleById(moduleId);
```

Comment:
will get an exception if not `FaceToFaceModule`, just saying - depends if we're fine with exceptions thrown from controller methods